src/Workspaces/VisualBasic/Portable/CodeGeneration/VisualBasicSyntaxGenerator.vb

```diff
@@ -248,6 +248,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.CodeGeneration
         End Function
 
         Public Overrides Function TypeExpression(typeSymbol As ITypeSymbol) As SyntaxNode
+            If typeSymbol.IsSystemVoid Then
```

Comment:
> Lastly, please note my comments in the PR about altering the approach altogether. I think it would be better if the abstract SyntaxGenerator checked if the return type is void and, if so, passes through null. Both the C# and VB syntax generators would then operate correctly. 

The alternate approach seems more sensible to me.  Right now changing "TypeExpression" is not ok as it will prevent being able to type *Legal Code*, and that's unacceptable IMO.

> Firstly, it's not technically possible to return System.Void from a function.

That may be true, but the purpose of our codegen APIs is not just to create legal code, but to create the code that was asked for :)    It's quite common that we may generate code that is not legal, but which is useful as an intermediate point as a user refactors what they have.

That said, have a 'void' returning method properly show up as a Sub for VB makes a ton of sense, and seems reasonable to have.  So i would go that way.